src/PortingAssistant.Client.Telemetry/TelemetryClient.cs

```diff
@@ -52,28 +52,31 @@ public Task<AmazonWebServiceResponse> SendAsync(TelemetryRequest request)
 
     public static class TelemetryClientFactory
     {
-        public static bool TryGetClient(string profile, TelemetryConfiguration config, out ITelemetryClient client, bool enabledDefaultCredentials = false)
-        {   
+        public static bool TryGetClient(string profile, TelemetryConfiguration config, out ITelemetryClient client,
+            bool enabledDefaultCredentials = false, AWSCredentials? awsCredentials = null)
+        {
             client = null;
-            if (string.IsNullOrEmpty(profile)) { return false; }
-            var chain = new CredentialProfileStoreChain();
-            AWSCredentials awsCredentials;
-            if (enabledDefaultCredentials)
+            if (awsCredentials == null)
             {
-                awsCredentials = FallbackCredentialsFactory.GetCredentials();
-                if (awsCredentials == null)
+                if (string.IsNullOrEmpty(profile)) { return false; }
+                var chain = new CredentialProfileStoreChain();
+                if (enabledDefaultCredentials)
                 {
-                    return false;
+                    awsCredentials = FallbackCredentialsFactory.GetCredentials();
+                    if (awsCredentials == null)
+                    {
+                        return false;
+                    }
                 }
-            }
-            else
-            {
-                if (!chain.TryGetAWSCredentials(profile, out awsCredentials))
+                else
                 {
-                    return false;
+                    if (!chain.TryGetAWSCredentials(profile, out awsCredentials))
+                    {
+                        return false;
+                    }
                 }
```

Comment:
I think you need to instantiate the client in this branch of the conditional if we are able to get credentials

Reply:
fix bugs in a new commit and add new tests.